src/main/java/com/ericsson/ei/frontend/utils/EIRequestsControllerUtils.java

```diff
@@ -90,24 +90,40 @@ private String getBackEndUrl(String requestQuery) throws EiBackendInstancesExcep
             List<NameValuePair> params = getParameters(requestQuery);
             if (requestQuery.contains(BACKEND_URL_KEY_NAME)) {
                 requestUrl = extractUrlFromParameters(params);
-            } else {
-                String backEndName = extractBackEndNameFromParameters(params);
-                BackEndInformation backEndInformation = backEndInstancesUtils.getBackEndInformationByName(backEndName);
-                if (backEndInformation == null) {
-                    throw new EiBackendInstancesException(
-                            "EI Backend instance '" + backEndName +  "' does not exist in EI Backend Instances list.");
+            } else if(requestQuery.contains(BACKEND_NAME_KEY_NAME)) {
+                try {
+                    requestUrl = getUrlFromBackendInstancesUtils(params);
+                } catch (EiBackendInstancesException e) {
+                    //Try to get default backend information if the requested one does not exist
+                    requestUrl = getDefaultBackendInstanceUrl();
                 }
-                requestUrl = backEndInformation.getUrlAsString();
+
+            } else {
+                requestUrl = getDefaultBackendInstanceUrl();
             }
         } else {
-            BackEndInformation backEndInformation = backEndInstancesUtils.getBackEndInformationByName(null);
-            requestUrl = backEndInformation.getUrlAsString();
+            requestUrl = getDefaultBackendInstanceUrl();
         }
 
         LOG.debug("Forwarding request to url '" + requestUrl + "'.");
         return requestUrl;
     }
 
+    private String getUrlFromBackendInstancesUtils(List<NameValuePair> params) throws EiBackendInstancesException {
+        String backEndName = extractBackEndNameFromParameters(params);
+        BackendInstance backendInstance = backEndInstancesUtils.getBackendInstance(backEndName);
+        if (backendInstance == null) {
+            throw new EiBackendInstancesException(
+                    "EI Backend instance '" + backEndName +  "' does not exist in EI Backend Instances list.");
+        }
+        return backendInstance.getUrlAsString();
```

Comment:
I would prefer getUrl().toString()
There is 2 operations in one method.

Reply:
This is functionality from before as well, i dont want to refactor it now either.